dispatches/models/renewables_case/wind_power.py

```diff
@@ -0,0 +1,149 @@
+# Import Pyomo libraries
+from pyomo.environ import Var, Param, NonNegativeReals, units as pyunits
+from pyomo.network import Port
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import (Component,
+                        ControlVolume0DBlock,
+                        declare_process_block_class,
+                        EnergyBalanceType,
+                        MomentumBalanceType,
+                        MaterialBalanceType,
+                        UnitModelBlockData,
+                        useDefault)
+from idaes.core.util.config import list_of_floats
+import idaes.logger as idaeslog
+
+import PySAM.Windpower as wind
+
+_log = idaeslog.getLogger(__name__)
+
+
+# TODO: move to idaes/core/util/config.py
+def list_of_list_of_floats(arg):
+    '''Domain validator for lists of floats
+
+    Args:
+        arg : argument to be cast to list of floats and validated
+
+    Returns:
+        List of list of floats
+    '''
+    try:
+        lst = [[float(i) for i in j] for j in arg]
+    except TypeError:
+        lst = [list_of_floats(arg), ]
+    return lst
+
+
+def dict_of_list_of_list_of_floats(arg):
+    dic = dict()
+    for k, v in arg.items():
+        dic[k] = list_of_list_of_floats(v)
+    return dic
+
+
+@declare_process_block_class("Wind_Power", doc="Wind plant using turbine powercurve and resource data")
+class WindpowerData(UnitModelBlockData):
+    """
+    Wind plant using turbine powercurve and resource data.
+    Unit model to convert wind resource into electricity.
+    """
+    CONFIG = ConfigBlock()
+    CONFIG.declare("dynamic", ConfigValue(
+        domain=In([False]),
+        default=False,
+        description="Dynamic model flag - must be False",
+        doc="""Wind plant does not support dynamic models, thus this must be False."""))
+    CONFIG.declare("has_holdup", ConfigValue(
+        default=False,
+        domain=In([False]),
+        description="Holdup construction flag",
+        doc="""Wind plant does not have defined volume, thus this must be False."""))
+    CONFIG.declare("resource_probability_density", ConfigValue(
+        default=None,
+        domain=dict_of_list_of_list_of_floats,
+        description="Dictionary of Time: List of (wind meters per sec, wind degrees clockwise from north, probability)",
+        doc="For each time in flowsheet's time set, a probability density function of "
+            "Wind speed [m/s] and Wind direction [degrees clockwise from N]"))
+
+    def build(self):
+        """Building model
+
+        Args:
+            None
+        Returns:
+            None
+        """
+        super(WindpowerData, self).build()
```

Comment:
Minor comment: IDAES only supports Python 3, so we can just do `super().build()`

Reply:
Fixed